ORBIT/phases/design/electrical_export.py

```diff
@@ -492,32 +567,87 @@ def calc_topside_mass_and_cost(self):
+        if self.cable.cable_type == "HVDC-monopole":
+            self.topside_cost = _design.get("topside_design_cost", 294e6)
+        elif self.cable.cable_type == "HVDC-bipole":
+            self.topside_cost = _design.get("topside_design_cost", 476e6)
+        else:
+            self.topside_cost = _design.get("topside_design_cost", 107.3e6)
 
     def calc_onshore_cost(self):
-        """Minimum Cost of Onshore Substation Connection"""
+        """Minimum Cost of Onshore Substation Connection
+
+        Parameters
+        ----------
+        shunt_cost_rate : int | float
+        onshore_converter_cost: int | float
+        switchgear_cost: int | float
+        """
+
+        _design = self.config.get("onshore_substation_design", {})
+
+        _shunt_cost_rate = _design.get("shunt_cost_rate", 1.3e4)  # per cable
+        _switchgear_cost = _design.get("switchgear_cost", 9.33e6)  # per cable
+        _compensation_rate = _design.get(
+            "compensation_rate", 31.3e6
+        )  # per cable
+
+        self.onshore_shunt_reactor_cost = (
+            self.compensation * self.num_cables * _shunt_cost_rate
+        )
+
+        if self.cable.cable_type == "HVDC-monopole":
+            self.onshore_converter_cost = (
+                self.num_substations
+                * self._design.get("onshore_converter_cost", 157e6)
+            )
+            self.onshore_switchgear_cost = 0
+            self.onshore_construction = self.num_substations * _design.get(
+                "onshore_construction_rate", 87.3e6
+            )
+            self.onshore_compensation_cost = 0
+
+        elif self.cable.cable_type == "HVDC-bipole":
+            self.onshore_converter_cost = (
+                self.num_substations
+                * self._design.get("onshore_converter_cost", 350e6)
+            )
+            self.onshore_switchgear_cost = 0
+            self.onshore_construction = self.num_substations * _design.get(
+                "onshore_construction_rate", 100e6
+            )
+            self.onshore_compensation_cost = 0
+
+        else:
+            self.onshore_converter_cost = 0
+            self.onshore_switchgear_cost = self.num_cables * _switchgear_cost
+            self.onshore_construction = self.num_substations * _design.get(
+                "onshore_construction_rate", 5e6
+            )
+            self.onshore_compensation_cost = (
+                self.num_cables * _compensation_rate
+                + self.onshore_shunt_reactor_cost
+            )
 
         self.onshore_cost = (
-            self.converter_cost
-            + self.dc_breaker_cost
+            self.onshore_converter_cost
+            + self.onshore_switchgear_cost
+            + self.onshore_construction
+            + self.onshore_compensation_cost
             + self.mpt_cost
-            + self.switchgear_cost
         )
 
         self._outputs["export_system"][
-            "onshore_construction_cost"
+            "onshore_substation_costs"
```

Comment:
Is this naming going to be backwards compatible with existing ORBIT users, or will it break things?

Reply:
Yes. I believe it is now but let me revisit it to confirm. By changing it to substation_costs, we avoid an issue in `export.py`

https://github.com/WISDEM/ORBIT/blob/8eb149eeb2926a98a813c73eb9aec40385009724/ORBIT/phases/install/cable_install/export.py#L174-L180

There is some existing code that counts the transmissions construction that would get overwritten if we specified these costs as "onshore_construction_cost". I'll make sure that is what we want.

Reply:
Goes with the comment above. There is some clean up to be had around the onshore cost before v2.0. However, I did check that nothing is double counted and made sure that it is backwards compatible.